mixer/adapter/kubernetesenv/kubernetesenv_test.go

```diff
@@ -384,6 +393,101 @@ func TestKubegen_Generate(t *testing.T) {
 	}
 }
 
+func createMultiClusterSecret(k8s *fake.Clientset) error {
+	data := map[string][]byte{}
+	secret := v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testSecretName,
+			Namespace: testSecretNameSpace,
+			Labels: map[string]string{
+				"istio/multiCluster": "true",
+			},
+		},
+		Data: map[string][]byte{},
+	}
+
+	data["testRemoteCluster"] = []byte("Test")
+	secret.Data = data
+	_, err := k8s.CoreV1().Secrets(testSecretNameSpace).Create(&secret)
+	return err
+}
+
+func deleteMultiClusterSecret(k8s *fake.Clientset) error {
+	var immediate int64
+
+	return k8s.CoreV1().Secrets(testSecretNameSpace).Delete(
+		testSecretName, &metav1.DeleteOptions{GracePeriodSeconds: &immediate})
+}
+
+func mockLoadKubeConfig(kubeconfig []byte) (*clientcmdapi.Config, error) {
+	return &clientcmdapi.Config{}, nil
+}
+
+func mockCreateInterfaceFromClusterConfig(clusterConfig *clientcmdapi.Config) (kubernetes.Interface, error) {
+	return fake.NewSimpleClientset(), nil
+}
+
+func verifyControllers(b *builder, controller_count int) bool {
+	var i int
+
+	// Sleep to allow the created/deleted multi-cluster secret to trigger the
+	// secret controller to callback the registered function to perform the
+	// expected action (add/delete a remote controller).
+	for i = 0; i < maxTestSleep; i++ {
```

Comment:
instead of this loop, could we use something like the existing `pkg/test/eventually.go` stuff? I'm not generally a huge fan of that pkg, but I think it might be prudent to reuse it here:

```
test.Eventually(t, "controllers wait", func() bool {
  return len(b.controllers) == controllerCount
})
```

Reply:
I implemented using test.Eventually. Added timeout values, otherwise a failed condition was taking 10 minutes to run test. Also note that this wait condition is also acting as the test.